src/main/kotlin/com/openlattice/subscriptions/PostgresSubscriptionService.kt

```diff
@@ -0,0 +1,160 @@
+package com.openlattice.subscriptions
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.openlattice.authorization.Principal
+import com.openlattice.authorization.PrincipalType
+import com.openlattice.graph.NeighborhoodQuery
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.postgres.PostgresTable.SUBSCRIPTIONS
+import com.openlattice.postgres.PostgresTable.SUBSCRIPTION_CONTACTS
+import com.openlattice.postgres.ResultSetAdapters
+import com.openlattice.postgres.streams.PostgresIterable
+import com.openlattice.postgres.streams.StatementHolder
+import com.zaxxer.hikari.HikariDataSource
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.time.OffsetDateTime
+import java.util.*
+import java.util.function.Function
+import java.util.function.Supplier
+
+class PostgresSubscriptionService(
+        private val hds: HikariDataSource,
+        private val mapper: ObjectMapper
+) : SubscriptionService {
+
+    // v code v
+
+    override fun createOrUpdateSubscription(subscription: NeighborhoodQuery, organizationId: UUID, user: Principal) {
+        hds.connection.use { conn ->
+            subscription.ids.map { ekid ->
```

Comment:
Would probably be good to refactor this and other calls in this class to do bulk writes of rows rather than each row individually